Sources/ApolloCodegenLib/ApolloCodegenConfiguration.swift

```diff
@@ -0,0 +1,140 @@
+    public let schema: URL
+    /// Glob of files to search for GraphQL operations. This should be used to find queries and any client schema extensions. Defaults to `./**/*.graphql`, which will search for `.graphql` files throughout all subfolders of the folder where the script is run.
+    public let glob: String
+    /// [optional] Path to an operation id JSON map file. If specified, also stores the operation IDs (hashes) as properties on operation types. Defaults to `nil`.
+    public let operationIDs: URL?
+  }
+
+  /// Configure the folder structure of the generated files.
+  public struct FileOutput {
+    /// The location structure for the schema types files.
+    public let schemaTypes: SchemaTypesFileOutput
+    /// The location structure for the operation object files.
+    public let operations: OperationsFileOutput
+  }
+
+  /// Defines the location structure for the schema types files.
+  public struct SchemaTypesFileOutput {
+    /// Compatible dependency manager automation.
+    public enum DependencyManagerAutomation {
+      /// No automation. The generated schema types files must be manually added to the main application target.
+      case includedInTarget
+      /// Generate a podspec file.
+      case cocoaPods
+      /// Generate a cartfile.
+      case carthage
+      /// Generate a package.swift file.
+      case swiftPackageManager
+    }
+
+    /// The namespace to use for generated operation objects. When combined with `.includedInTarget` this should be the main target name. When used with any other dependency manager this should be the name of the new shared module that will be created.
+    public let name: String
+    /// An absolute location where the schema types files should be generated.
+    public let url: URL
+    /// Automation to ease the integration of the generated schem types file with compatible dependency managers.
+    public let dependencyAutomation: DependencyManagerAutomation
+  }
+
+  /// Defines the location structure for the operation object files.
+  public enum OperationsFileOutput {
+    /// All operation object files will be located in the module with the schema types.
+    case inSchemaModule
+    /// Operation object files will be co-located relative to the defining `.graphql` file.
+    case relative(subpath: String?)
+    /// All operation object files will be located in the specified path.
+    case absolute(url: URL)
+  }
+
+  /// Specify the formatting of the GraphQL query string literal.
+  public enum GraphQLQueryStringLiteralFormat {
+    /// The query string will be copied into the operation object with all line break formatting removed.
+    case singleLine
+    /// The query string will be copied with original formatting into the operation object.
+    case multiline
+    /// The query string will not be copied into the operation object. You must be using preregisterd queries with this configuration.
+    case exclude
```

Comment:
I'm not a fan of this here. It means someone could create an invalid configuration if then aren't using APQs and put `exclude`. I think that this should just be part of the APQ configuration, and if it's persisted queries only, then we just ignore this option since we aren't generating the query string literals.

Reply:
I agree with @AnthonyMDev here too - I think the question for this enum is more "If there's a string literal, how should we format it" rather than "should we include the query string"

Reply:
Great suggestion, thanks!